chains/substrate/calls/pallets/bridge/bridge.go

```diff
@@ -0,0 +1,133 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: BUSL-1.1
+
+package bridge
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+
+	"github.com/ChainSafe/sygma-relayer/chains/substrate/connection"
+
+	"github.com/ChainSafe/sygma-relayer/chains/substrate/client"
+	"github.com/ChainSafe/sygma-relayer/chains/substrate/executor/proposal"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/signer/core/apitypes"
+	"github.com/rs/zerolog/log"
+)
+
+type BridgeProposal struct {
+	OriginDomainID uint8
+	ResourceID     [32]byte
+	DepositNonce   uint64
+	Data           []byte
+}
+
+type BridgePallet struct {
+	client *client.SubstrateClient
+}
+
+func NewBridgePallet(
+	client *client.SubstrateClient,
+) *BridgePallet {
+	return &BridgePallet{
+		client: client,
+	}
+}
+
+func (p *BridgePallet) ExecuteProposals(
+	conn *connection.Connection,
+	proposals []*proposal.Proposal,
+	signature []byte,
+) (*types.Hash, error) {
+	bridgeProposals := make([]BridgeProposal, 0)
+	for _, prop := range proposals {
+		bridgeProposals = append(bridgeProposals, BridgeProposal{
+			OriginDomainID: prop.Source,
+			DepositNonce:   prop.DepositNonce,
+			ResourceID:     prop.ResourceId,
+			Data:           prop.Data,
+		})
+	}
+
+	return p.client.Transact(
+		conn,
+		"execute_proposal",
+		bridgeProposals,
+		signature,
+	)
+}
+
+func (p *BridgePallet) ProposalsHash(proposals []*proposal.Proposal) ([]byte, error) {
```

Comment:
Probably should tests this with unit tests

Reply:
Also, I know I added this to EVM contracts, but it seems in the wrong place as it does not do any call to pallet so please can you think if there is some other more usefull place to put it
Also, is proposals hash the same across all networks? If it is (I think it might be) we should probably put it in the proposal package